libs/wire-api/src/Wire/API/Conversation.hs

```diff
@@ -247,6 +249,26 @@ instance FromJSON a => FromJSON (ConversationList a) where
       <$> o A..: "conversations"
       <*> o A..: "has_more"
 
+-- | Used on the POST /list-conversations endpoint
+-- TODO: add to golden tests
+data ListConversations = ListConversations
+  { lQualifiedIds :: Maybe (NonEmpty (Qualified ConvId)),
+    lStartId :: Maybe (Qualified ConvId),
+    lSize :: Maybe (Range 1 500 Int32)
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving (FromJSON, ToJSON, S.ToSchema) via Schema ListConversations
+
+instance ToSchema ListConversations where
+  schema =
+    objectWithDocModifier
+      "ListConversations"
+      (description ?~ "A request to list some or all of a user's conversations, including remote ones")
+      $ ListConversations
+        <$> lax (lQualifiedIds .= opt (field "qualified_ids" (nonEmptyArray schema)))
+        <*> lax (lStartId .= opt (field "start_id" schema))
+        <*> lax (lSize .= opt (field "size" schema))
```

Comment:
No need to use `lax` here. `lax` is important when `opt` / `optWithDefault` is used in the inner schema. If you use `opt` on the `field` schema itself, it already accepts a missing field, so `lax` does not do anything.

In fact, after #1621, it is better to just use `optField` here, as in:
```haskell
lQualifiedIds .= optField "qualified_ids" (nonEmptyArray schema)
```
and similarly for the other ones. The difference is that if we get as input a JSON like:
```json
{ "qualified_ids": ["invalid"] }
```
with `opt` we just get `Nothing` for that field, while with `optField` we get an error.

Reply:
Thanks! I changed this to use optField. Have you come across places where we don't want to omit the field when generating json though? Otherwise the extra argument to optField (Nothing / Aeson.null in the usages so far) could be simplified to always use Nothing.